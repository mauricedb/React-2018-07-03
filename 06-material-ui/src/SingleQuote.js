import React from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

const SingleQuote = ({ joke }) => {
  return (
    <Card style={{ width: 200, margin: 20 }}>
      <CardContent>
        <Typography>{joke.joke}</Typography>
      </CardContent>
    </Card>
  );
};

export default SingleQuote;
