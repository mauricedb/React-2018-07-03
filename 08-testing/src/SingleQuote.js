import React from 'react';

const SingleQuote = ({ joke }) => {
  return <li>{joke.joke}</li>;
};

export default SingleQuote;
