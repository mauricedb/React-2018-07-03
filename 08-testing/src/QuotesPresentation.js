import React, { Component } from 'react';
import propTypes from 'prop-types';
import SingleQuote from './SingleQuote';

class QuotesPresentation extends Component {
  static propTypes = {
    jokes: propTypes.array.isRequired,
    loading: propTypes.bool,
    error: propTypes.object
  };

  static defaultProps = {
    jokes: []
  };

  render() {
    const { jokes, loading, error } = this.props;

    if (loading) {
      return 'Loading';
    }

    if (error) {
      return <div style={{ color: 'red' }}>{error.message}</div>;
    }

    return (
      <div onClick={() => {}}>
        <h1>Chuck Norris Quotes.</h1>
        There are {jokes.length} jokes
        <ul>{jokes.map(joke => <SingleQuote key={joke.id} joke={joke} />)}</ul>
      </div>
    );
  }
}

export default QuotesPresentation;
