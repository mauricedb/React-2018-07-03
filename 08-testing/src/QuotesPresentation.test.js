import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import QuotesPresentation from './QuotesPresentation';
import SingleQuote from './SingleQuote';

describe('The QuotesPresentation component', () => {
  describe('unit tests', () => {
    it('Can render', () => {
      const wrapper = shallow(<QuotesPresentation />);
    });

    it('Can render 2 jokes', () => {
      const wrapper = shallow(
        <QuotesPresentation jokes={[{ id: 1 }, { id: 2 }]} />
      );

      const jokes = wrapper.find(SingleQuote);
      expect(jokes.length).toBe(2);
    });
  });

  describe('intergration tests', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <QuotesPresentation jokes={[{ id: 1, joke: 'The joke' }]} />
      );
    });

    it('Can render a joke', () => {
      const jokes = wrapper.find(SingleQuote);
      const joke = jokes.dive();
      expect(joke.text()).toBe('The joke');
    });

    // for (let i = 0; i < 3; i++) {
    //   it(`Can render ${i} jokes`, () => {
    //     const jokes = wrapper.find(SingleQuote);
    //     expect(jokes.length).toBe(i);
    //   });
    // }
  });

  test('Link renders correctly', () => {
    const tree = renderer
      .create(<QuotesPresentation jokes={[{ id: 1, joke: 'The joke' }]} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('The quotes', () => {
    const quotes = [{ id: 1 }, { id: 2 }, { id: 3 }];
    expect(quotes).toMatchSnapshot();
  });
});
