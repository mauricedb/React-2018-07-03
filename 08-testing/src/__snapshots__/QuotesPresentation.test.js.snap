// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The QuotesPresentation component Link renders correctly 1`] = `
<div
  onClick={[Function]}
>
  <h1>
    Chuck Norris Quotes.
  </h1>
  There are 
  1
   jokes
  <ul>
    <li>
      The joke
    </li>
  </ul>
</div>
`;

exports[`The QuotesPresentation component The quotes 1`] = `
Array [
  Object {
    "id": 1,
  },
  Object {
    "id": 2,
  },
  Object {
    "id": 3,
  },
]
`;
